connect/runtime/src/test/java/org/apache/kafka/connect/storage/KafkaOffsetBackingStoreTest.java

```diff
@@ -329,16 +308,9 @@ public void onCompletion(Throwable error, Void result) {
         assertTrue(invoked.get());
 
         // Getting data should read to end of our published data and return it
-        final AtomicBoolean secondGetInvokedAndPassed = new AtomicBoolean(false);
-        store.get(Arrays.asList(null, TP1_KEY), new Callback<Map<ByteBuffer, ByteBuffer>>() {
-            @Override
-            public void onCompletion(Throwable error, Map<ByteBuffer, ByteBuffer> result) {
-                assertEquals(TP0_VALUE, result.get(null));
-                assertNull(result.get(TP1_KEY));
-                secondGetInvokedAndPassed.set(true);
-            }
-        }).get(10000, TimeUnit.MILLISECONDS);
-        assertTrue(secondGetInvokedAndPassed.get());
+        Map<ByteBuffer, ByteBuffer> offsets = store.get(Arrays.asList(null, TP1_KEY)).get(10000, TimeUnit.MILLISECONDS);
+        assertEquals(TP0_VALUE, offsets.get(null));
+        assertNull(offsets.get(TP1_KEY));
```

Comment:
The bad thing about removing the callback parameter is that these tests no longer have a good way of known whether the reader completed successfully. Do we really not need this anymore?

Reply:
I could be wrong, but isn't it enough to just verify that the call to `ConvertingFutureCallback::get` returned without throwing an exception? Any additional testing logic contained in the callback (such as in https://github.com/apache/kafka/pull/7532/files#diff-d363e0391be734af7583fd6c3a30d77eL256-L260) is retained by examining the offsets returned from that call (such as in https://github.com/apache/kafka/pull/7532/files#diff-d363e0391be734af7583fd6c3a30d77eR247-R248).

Retaining this kind of API (even if it's completely internal) just for testing purposes doesn't seem like a great idea; I'd prefer to remove it if it's unneeded and perform the testing logic separately. I believe I've accomplished this but if it appears I've missed anything it's not out of the question to leave it as-is for now.

Reply:
You raise good points about how the existing checks can be done on the result of the method invocation, and since we've mocked the CloseableOffsetStorageReader implementation we do know what we returned. I agree that in general this seemed like a complication that was only used in testing, and that we have not lost any visibility within the tests. Thanks for the quick response.